packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -538,63 +661,55 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
   List<Widget> _createTiles() {
     final MaterialLocalizations localizations = MaterialLocalizations.of(context);
     assert(localizations != null);
-    final List<Widget> children = <Widget>[];
+
+    final ThemeData themeData = Theme.of(context);
+
+    Color themeColor;
+    switch (themeData.brightness) {
+      case Brightness.light:
+        themeColor = themeData.primaryColor;
+        break;
+      case Brightness.dark:
+        themeColor = themeData.accentColor;
+        break;
+    }
+
+    ColorTween colorTween;
     switch (widget.type) {
       case BottomNavigationBarType.fixed:
-        final ThemeData themeData = Theme.of(context);
-        final TextTheme textTheme = themeData.textTheme;
-        Color themeColor;
-        switch (themeData.brightness) {
-          case Brightness.light:
-            themeColor = themeData.primaryColor;
-            break;
-          case Brightness.dark:
-            themeColor = themeData.accentColor;
-            break;
-        }
-        final ColorTween colorTween = ColorTween(
-          begin: textTheme.caption.color,
-          end: widget.fixedColor ?? themeColor,
+        colorTween = ColorTween(
+          begin: widget.unselectedItemColor ?? themeData.textTheme.caption.color,
+          end: widget.selectedItemColor ?? widget.fixedColor ?? themeColor,
         );
-        for (int i = 0; i < widget.items.length; i += 1) {
-          children.add(
-            _BottomNavigationTile(
-              widget.type,
-              widget.items[i],
-              _animations[i],
-              widget.iconSize,
-              onTap: () {
-                if (widget.onTap != null)
-                  widget.onTap(i);
-              },
-              colorTween: colorTween,
-              selected: i == widget.currentIndex,
-              indexLabel: localizations.tabLabel(tabIndex: i + 1, tabCount: widget.items.length),
-            ),
-          );
-        }
         break;
       case BottomNavigationBarType.shifting:
-        for (int i = 0; i < widget.items.length; i += 1) {
-          children.add(
-            _BottomNavigationTile(
-              widget.type,
-              widget.items[i],
-              _animations[i],
-              widget.iconSize,
-              onTap: () {
-                if (widget.onTap != null)
-                  widget.onTap(i);
-              },
-              flex: _evaluateFlex(_animations[i]),
-              selected: i == widget.currentIndex,
-              indexLabel: localizations.tabLabel(tabIndex: i + 1, tabCount: widget.items.length),
-            ),
-          );
-        }
+        colorTween = ColorTween(
+          begin: widget.unselectedItemColor ?? Colors.white,
+          end: widget.selectedItemColor ?? Colors.white,
+        );
         break;
     }
-    return children;
+
+    return List<Widget>.generate(widget.items.length, (int i) {
```

Comment:
Generally prefer to manually construct a list, even if it is only marginally faster

Reply:
Didn't realize it was slower to use a List generator, switched to a for-loop.